velox/functions/lib/Re2Functions.cpp

```diff
@@ -805,71 +880,95 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> re2ExtractSignatures() {
   };
 }
 
-std::pair<PatternKind, vector_size_t> determinePatternKind(StringView pattern) {
-  vector_size_t patternLength = pattern.size();
-  vector_size_t i = 0;
-  // Index of the first % or _ character.
-  vector_size_t wildcardStart = -1;
-  // Index of the first character that is not % and not _.
-  vector_size_t fixedPatternStart = -1;
-  // Total number of % characters.
-  vector_size_t anyCharacterWildcardCount = 0;
-  // Total number of _ characters.
-  vector_size_t singleCharacterWildcardCount = 0;
-  auto patternStr = pattern.data();
+PatternParameters determinePatternKind(const StringView& pattern) {
+  auto i = 0;
+  const auto patternLength = pattern.size();
+  const auto patternString = pattern.str();
+  auto wildcardStart = 0;
+  auto wildcardCount = 0;
+  auto anyCharacterWildcardCount = 0;
+  auto singleCharacterWildcardCount = 0;
+  auto fixedPatternStart = 0;
+  auto fixedPatternCount = 0;
+  std::vector<std::string> fixedPatterns;
 
   while (i < patternLength) {
-    if (patternStr[i] == '%' || patternStr[i] == '_') {
-      // Ensures that pattern has a single contiguous stream of wildcard
-      // characters.
-      if (wildcardStart != -1) {
-        return std::make_pair(PatternKind::kGeneric, 0);
-      }
+    if (patternString[i] == '%' || patternString[i] == '_') {
       // Look till the last contiguous wildcard character, starting from this
       // index, is found, or the end of pattern is reached.
       wildcardStart = i;
       while (i < patternLength &&
-             (patternStr[i] == '%' || patternStr[i] == '_')) {
-        singleCharacterWildcardCount += (patternStr[i] == '_');
-        anyCharacterWildcardCount += (patternStr[i] == '%');
+             (patternString[i] == '%' || patternString[i] == '_')) {
+        singleCharacterWildcardCount += (patternString[i] == '_');
+        anyCharacterWildcardCount += (patternString[i] == '%');
         i++;
       }
-    } else {
-      // Ensure that pattern has a single fixed pattern.
-      if (fixedPatternStart != -1) {
-        return std::make_pair(PatternKind::kGeneric, 0);
+      if (!singleCharacterWildcardCount) {
```

Comment:
It would be good to delay this until much later in the code towards the part where its being used.

Reply:
Thanks for the suggestion, because this keeps a track of the number of wildcard character '%' streams, it needs to be here and moving this outside the loop causes tests to fail. Removed the check though since it can be added subsequently.